Sources/Datadog/RUM/DataModels/RUMDataModels.swift

```diff
@@ -0,0 +1,809 @@
+        case connectivity = "connectivity"
+        case date = "date"
+        case session = "session"
+        case usr = "usr"
+        case view = "view"
+        case action = "action"
+        case longTask = "long_task"
+        case type = "type"
+    }
+}
+
+/// Action properties
+// MARK: - RUMLongTaskAction
+internal struct RUMLongTaskAction: Codable {
+    /// UUID of the action
+    let id: String
+
+    enum CodingKeys: String, CodingKey {
+        case id = "id"
+    }
+}
+
+/// Long Task properties
+// MARK: - RUMLongTaskLongTask
+internal struct RUMLongTaskLongTask: Codable {
+    /// Duration in ns of the long task
+    let duration: Int64
+
+    enum CodingKeys: String, CodingKey {
+        case duration = "duration"
+    }
+}
+
+/// Schema of all properties of a Resource event
+///
+/// Schema of common properties of RUM events
+// MARK: - RUMResource
+internal struct RUMResource: RUMDataModel {
+    /// Internal properties
+    let dd: RUMActionDD
+    /// Application properties
+    let application: RUMApplication
+    /// Device connectivity properties
+    let connectivity: RUMConnectivity?
+    /// Start of the event in ms from epoch
+    let date: Int64
+    /// Session properties
+    let session: RUMSession
+    /// User properties
+    let usr: RUMUSR?
+    /// View properties
+    let view: RUMActionView
+    /// Action properties
+    let action: RUMResourceAction?
+    /// Resource properties
+    let resource: RUMResourceResource
+    /// RUM event type
+    let type = "resource"
+
+    enum CodingKeys: String, CodingKey {
+        case dd = "_dd"
+        case application = "application"
+        case connectivity = "connectivity"
+        case date = "date"
+        case session = "session"
+        case usr = "usr"
+        case view = "view"
+        case action = "action"
+        case resource = "resource"
+        case type = "type"
+    }
+}
+
+/// Action properties
+// MARK: - RUMResourceAction
+internal struct RUMResourceAction: Codable {
+    /// UUID of the action
+    let id: String
+
+    enum CodingKeys: String, CodingKey {
+        case id = "id"
+    }
+}
+
+/// Resource properties
+// MARK: - RUMResourceResource
+internal struct RUMResourceResource: Codable {
+    /// Connect phase properties
+    let connect: RUMConnect?
+    /// DNS phase properties
+    let dns: RUMDNS?
+    /// Download phase properties
+    let download: RUMDownload?
+    /// Duration of the resource
+    let duration: Int64
+    /// First Byte phase properties
+    let firstByte: RUMFirstByte?
+    /// HTTP method of the resource
+    let method: RUMMethod?
+    /// Redirect phase properties
+    let redirect: RUMRedirect?
+    /// Size in octet of the resource response body
+    let size: Int64?
+    /// SSL phase properties
+    let ssl: RUMSSL?
+    /// HTTP status code of the resource
+    let statusCode: Int64?
+    /// Resource type
+    let type: RUMResourceType
+    /// URL of the resource
+    let url: String
```

Comment:
Is it easy doable to sort these fields, i.e. non-optional first or just follow the order they are defined in `rum-events-format`? It's a bit weird that the most important ones (`type` and `url`) are defined at the end, while less important / optional (`dns`, `firstByte`) are at the top of initializer. The `rum-events-format` lists important things first.

Reply:
it follows the ordering in json-schemas... but in reversed
let me see this one too how i can reverse it